homeassistant/components/devolo_home_control/binary_sensor.py

```diff
@@ -0,0 +1,73 @@
+"""Platform for binary sensor integration."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN
+from .devolo_device import DevoloDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Get all binary sensor and multi level sensor devices and setup them via config entry."""
+    entities = []
+
+    for device in hass.data[DOMAIN]["homecontrol"].binary_sensor_devices:
+        for binary_sensor in device.binary_sensor_property:
+            entities.append(
+                DevoloBinaryDeviceEntity(
+                    homecontrol=hass.data[DOMAIN]["homecontrol"],
+                    device_instance=device,
+                    element_uid=binary_sensor,
+                )
+            )
+
+    async_add_entities(entities, False)
+
+
+class DevoloBinaryDeviceEntity(DevoloDeviceEntity, BinarySensorEntity):
+    """Representation of a binary sensor within devolo Home Control."""
+
+    def __init__(self, homecontrol, device_instance, element_uid):
+        """Initialize a devolo binary sensor."""
+        super().__init__(
+            homecontrol=homecontrol,
+            device_instance=device_instance,
+            element_uid=element_uid,
+            name=device_instance.itemName
+            + " "
+            + device_instance.binary_sensor_property.get(element_uid).sub_type
+            if device_instance.binary_sensor_property.get(element_uid).sub_type != ""
+            else device_instance.itemName
+            + " "
+            + device_instance.binary_sensor_property.get(element_uid).sensor_type,
```

Comment:
Please use an `if` block when a `Ternary` is longer than a single line

Reply:
I've changed it in the latest commit. Is this like you want it?

Reply:
I gave a more detailed example below. HTH

Reply:
Yes, this helped. Thanks

Done in the latest commit.